tracer/test/Datadog.Trace.TestHelpers/ProfilerHelper.cs

```diff
@@ -30,6 +36,61 @@ public class ProfilerHelper
             // clear all relevant environment variables to start with a clean slate
             EnvironmentHelper.ClearProfilerEnvironmentVariables();
 
+            // this is nasty, but it's the only way I could find to force
+            // a .NET Framework exe to run in 32 bit if required
+            if (EnvironmentTools.IsWindows()
+             && !EnvironmentHelper.IsCoreClr()
+             && !EnvironmentTools.IsTestTarget64BitProcess()
+             && Path.GetExtension(executable) == ".exe"
+             && !ExesToExcludeFromCorflags.Contains(Path.GetFileNameWithoutExtension(executable)))
+            {
+                string corFlagsExe = null;
+                var programFiles = Environment.GetFolderPath(Environment.SpecialFolder.ProgramFilesX86);
+                var dotnetWindowsSdkToolsFolder = Path.Combine(programFiles, "Microsoft SDKs", "Windows", "v10.0A", "bin");
+
+                if (Directory.Exists(dotnetWindowsSdkToolsFolder))
+                {
+                    // get sub directories, e.g.
+                    // @"C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.8.1 Tools",
+                    // @"C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.8 Tools",
+                    foreach (var folder in Directory.EnumerateDirectories(dotnetWindowsSdkToolsFolder))
+                    {
+                        var exe = Path.Combine(folder, "x64", "CorFlags.exe");
+                        if (File.Exists(exe))
+                        {
+                            corFlagsExe = exe;
+                            break;
+                        }
+                    }
+                }
+
+                if (!string.IsNullOrEmpty(corFlagsExe))
+                {
+                    agent.Output?.WriteLine($"CorFlags.exe found at {corFlagsExe}");
+                    var setBit = EnvironmentTools.IsTestTarget64BitProcess()
```

Comment:
`EnvironmentTools.IsTestTarget64BitProcess() == false` here so I think we can unconditionally set the value to `/32BITREQ+`, unless we were to break this out into a separate function

Reply:
I moved this to a separate function.

Originally I was _always_ setting the x86/x64 flag, but I decided to switch to only doing it on `x86` to avoid the extra overhead. Given everyone (pretty much) will test x64 locally, that means no impact. Also, we don't ever test x64/x86 in parallel on the same machine (currently), and we always re-build the samples when switching between x64/x86 when running through Nuke.

The only situation in which _not_ explicitly setting the 32bit flag for x64 is if you do the following:

- Build using Nuke for x86
- Run tests in VS/Rider _without_ rebuilding the samples (defaults to x64)

That seems a pretty unlikely situation, but if we ever hit it, we can easily update this to force setting the flag for x64 bit executables too.